samples/nextjs/src/lib/sitemap-fetcher/plugins/disconnectedSitemapService.ts

```diff
@@ -0,0 +1,52 @@
+﻿import { DisconnectedSitemapService } from '@sitecore-jss/sitecore-jss-nextjs';
+// #START_EMPTY
+import { ManifestInstance } from '@sitecore-jss/sitecore-jss-dev-tools';
+import { SitemapFetcherPlugin } from '..';
+// #END_EMPTY
+
+class DisconnectedSitemapServicePlugin implements SitemapFetcherPlugin {
+  _disconnectedSitemapService: DisconnectedSitemapService;
+
+  constructor() {
+    // #START_EMPTY
+    this._disconnectedSitemapService = new DisconnectedSitemapService(
+      this.getManifest() as unknown as ManifestInstance
+    );
+    // #END_EMPTY
+  }
+
+  /**
+   * Get sitecore-import.json manifest
+   */
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  getManifest() {
+    if (process.env.JSS_MODE !== 'disconnected') return null;
+
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const manifest = require('sitecore/manifest/sitecore-import.json');
+
+      return manifest;
+    } catch (error) {
+      throw Error(
+        "[Disconnected Export] Please make sure you've started the disconnected proxy `npm run start:disconnected-proxy`"
+      );
+    }
+  }
+
+  async exec() {
+    // If we are in Export mode
+    if (process.env.EXPORT_MODE) {
+      // #START_EMPTY
+      // Disconnected Export mode
+      if (process.env.JSS_MODE === 'disconnected') {
+        return this._disconnectedSitemapService.fetchExportSitemap();
+      }
+      // #END_EMPTY
+    }
+
+    return [];
```

Comment:
Can clean this up quite a bit if empty flag is not a concern.
```suggestion
    // If we are in Disconnected Export mode
    if (process.env.JSS_MODE === 'disconnected' && process.env.EXPORT_MODE) {
      return this._disconnectedSitemapService.fetchExportSitemap();
    }
    return [];
```